### PR TITLE
Add go vendor support to the go build pipeline.

### PR DESCRIPTION
**Pull request by dlorenc**

### CODE REVIEW

File: pkg/build/pipelines/go/build.yaml [hunk 1]
```diff
@@ -71,6 +76,10 @@ pipeline:
           go get $dep
         done
         go mod tidy
+        # If vendor is specified, update the vendor directory
+        if [ "${{inputs.vendor}}" == "true" ]; then
```
Comment (kaniini) [inline pkg/build/pipelines/go/build.yaml:80]: Why not
```suggestion
        ${{inputs.vendor}} && go mod vendor
```

Comment (dlorenc) [reply pkg/build/pipelines/go/build.yaml:80]: ooooooh using bool/string conversion to our advantage, nice :)

File: pkg/build/pipelines/go/build.yaml [hunk 2]
```diff
@@ -24,6 +24,11 @@ inputs:
       the apk will be in prefix / install-dir / output
     required: true
 
+  vendor:
+    description: |
+      If true, the go mod command will also update htethe vendor directory
```
Comment (ajayk) [inline pkg/build/pipelines/go/build.yaml:29]: nit typo after the update `htethe`

**Comment by dlorenc** [comment]
Fixed and tested!

**Review by imjasonh — APPROVED** [review]

**Pull request merged**